In [1]:
import pandas as pd

In [2]:
purchase_data = pd.read_csv("Resources/purchase_data.csv")


In [3]:
purchase_data

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

# Player Count 

* Display the Total Number of Players 


In [4]:
total_players = len(purchase_data["SN"].unique())
print(f"Total number of players = {total_players}")

Total number of players = 576


# Purchasing Analysis (Total)





* Optional: give the displayed data cleaner formatting

* Display the summary data frame

In [5]:
#Run basic calculations to obtain number of unique items, average price, etc.
unique_purchases = len(purchase_data["Item Name"].unique())
print(f"Number of Unique Items = {unique_purchases}")

average_price = round((purchase_data["Price"].mean()),2)
print(f"Average price of each item  = {average_price}")

total_purchases = purchase_data["Purchase ID"].count()
print(f"Total number of purchases = {total_purchases}")

total_revenue = purchase_data["Price"].sum()
print(f"Total revenue = {total_revenue}")

Number of Unique Items = 179
Average price of each item  = 3.05
Total number of purchases = 780
Total revenue = 2379.77


### Create a summary data frame to hold the results

In [6]:

purchase_total_df = pd.DataFrame({
        "Number of Unique Items" :[179],
        "Average price of each item":[3.05],
        "Total number of purchases":[780],
        "Total revenue":[2379.77]
})
purchase_total_df

Number of Unique Items  Average price of each item  \
0                     179                        3.05   

   Total number of purchases  Total revenue  
0                        780        2379.77

# Gender Demographics



In [7]:
#Percentage and Count of Male Players
#Percentage and Count of Female Players
#Percentage and Count of Other / Non-Disclosed

gender_count = purchase_data.groupby("Gender")["SN"].unique()


male_count = len(gender_count["Male"])
female_count = len(gender_count["Female"])
other_count = len(gender_count["Other / Non-Disclosed"])

print(f"There are {male_count} male players, {female_count} female players and {other_count} players of another gender")

There are 484 male players, 81 female players and 11 players of another gender


In [8]:
#total number of male, female and other gendered players 


male = 484
female = 81
other = 11
total = male + female + other

male_percent = round(((male/total) * 100),2)
female_percent = round(((female/total) * 100),2)
other_percent = round(((other/total) * 100),2)

print(f"There are {female} female players, {male} male players and {other} players of another gender.\nThe total player base consist of {female_percent} females, {male_percent} males and and {other_percent} players of another gender.")

There are 81 female players, 484 male players and 11 players of another gender.
The total player base consist of 14.06 females, 84.03 males and and 1.91 players of another gender.


In [9]:
gender_count_df = pd.DataFrame({
        "Gender":["Male","Female","Other"],
        "Total Count":[male,female,other],
        "Percentage of players":[male_percent,female_percent,other_percent]
})
gender_count_df

Gender  Total Count  Percentage of players
0    Male          484                  84.03
1  Female           81                  14.06
2   Other           11                   1.91

# Purchasing Analysis (Gender)


In [10]:
#number of times  users purchased per gender 
print(purchase_data["Gender"].value_counts())

male_purchases_num = 652
female_purchases_num = 113
other_purchases_num = 15

Male                     652
Female                   113
Other / Non-Disclosed     15
Name: Gender, dtype: int64


In [11]:
#Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender

#groupby 
total_purchase_gender = purchase_data.groupby("Gender")["Price"].sum()
print(total_purchase_gender)

#calculating total amount spent for each gender group
male_total_purchase_value = total_purchase_gender["Male"]
female_total_purchase_value = total_purchase_gender["Female"]
other_total_purchase_value = total_purchase_gender["Other / Non-Disclosed"]

#calculating total amount spend on average per person in each gender group
male_ave_perp = round(male_total_purchase_value/male,2)
female_ave_perp = round(female_total_purchase_value/female,2)
other_ave_perp = round(other_total_purchase_value/other,2)

Gender
Female                    361.94
Male                     1967.64
Other / Non-Disclosed      50.19
Name: Price, dtype: float64


In [12]:
#calculating Purchase Count, Average Purchase Price, Total Purchase Value and Avg Total Purchase per 

male_ave_purchase = round(male_total_purchase_value/male_purchases_num,2)
female_ave_purchase = round(female_total_purchase_value/female_purchases_num,2)
other_ave_purchase =round(other_total_purchase_value/other_purchases_num,2)


In [13]:
#Create a summary data frame to hold the results
#Optional: give the displayed data cleaner formatting
#Display the summary data frame

pd.DataFrame({"Gender":["Male","Female","Other"],
              "Purchase count":[male_purchases_num,female_purchases_num,other_purchases_num],
              "Average Purchase Price":[male_ave_purchase,female_ave_purchase,other_ave_purchase],
              "Total Purchase Value":[male_total_purchase_value,female_total_purchase_value,other_total_purchase_value],
              "Average total purchase per gender":[male_ave_perp,female_ave_perp,other_ave_perp]
    
    
})

Gender  Purchase count  Average Purchase Price  Total Purchase Value  \
0    Male             652                    3.02               1967.64   
1  Female             113                    3.20                361.94   
2   Other              15                    3.35                 50.19   

   Average total purchase per gender  
0                               4.07  
1                               4.47  
2                               4.56

# Age Demographics




* Optional: round the percentage column to two decimal points


* Display Age Demographics Table

In [14]:
#Establish bins for ages
min_age = purchase_data["Age"].min()
max_age = purchase_data["Age"].max()
min_age,max_age

(7, 45)

In [15]:
bin = [0,9,14,19,24,29,34,39,50]
age_category = ["<10","10 to 14","15 to 19", "20 to 24", 
                "25 to 29", "30 to 34", "35 to 39", "40+"]

In [16]:
pd.cut(purchase_data["Age"], bin, labels = age_category)

0      20 to 24
1           40+
2      20 to 24
3      20 to 24
4      20 to 24
         ...   
775    20 to 24
776    20 to 24
777    20 to 24
778         <10
779    20 to 24
Name: Age, Length: 780, dtype: category
Categories (8, object): [<10 < 10 to 14 < 15 to 19 < 20 to 24 < 25 to 29 < 30 to 34 < 35 to 39 < 40+]

In [17]:
#Categorize the existing players using the age bins. Hint: use pd.cut()
purchase_data["Age category"] = pd.cut(purchase_data["Age"], bin, labels = age_category)

In [18]:
purchase_data

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price Age category  
0    Extraction, Quickblade Of Trembling Hands   3.53     20 to 24  
1                            Frenzied Scimitar   1.56          40+  
2                                 Final Critic   4.88     20 to 24  
3                                  Blindscythe   3.27     20 to 24  
4                                         Fury   1.44     20 to 24  
..                                         ...    ...          ...  
775                                       Wolf   3.54     20 to 24  
776                           Exiled Doomblade   1.63     20 to 24  
777      Celeste, Incarnation of the Corrupted   3.46     20 to 24  
778                               Final Critic   4.19          <10  
779                                       Dawn   4.60     20 to 24  

[780 rows x 8 columns]

In [19]:
#Calculate the numbers and percentages by age group
group_age_category = purchase_data.groupby("Age category")
age_cat_count = group_age_category["SN"].unique()
age_cat_count


Age category
<10         [Eusri44, Haillyrgue51, Seuthep89, Heudai45, T...
10 to 14    [Lirtossa84, Aeralria27, Reulae52, Zhisrisu83,...
15 to 19    [Idai61, Sondim73, Aidaillodeu39, Yaliru88, Ji...
20 to 24    [Lisim78, Ithergue48, Chamassasya86, Iskosia90...
25 to 29    [Lisirra87, Yasrisu92, Phaelap26, Chamirraya83...
30 to 34    [Chamalo71, Siala43, Idaisuir85, Iaralrgue74, ...
35 to 39    [Itheria73, Chanosian48, Saesrideu94, Siarithr...
40+         [Lisovynya38, Eyrian71, Jiskjask85, Isursuir31...
Name: SN, dtype: object

In [20]:
#number of players per age group
ten = len(age_cat_count["<10"])
fourteen = len(age_cat_count["10 to 14"])
nineteen = len(age_cat_count["15 to 19"])
twentyfour = len(age_cat_count["20 to 24"])
twentynine = len(age_cat_count["25 to 29"])
thirtyfour = len(age_cat_count["30 to 34"])
thirtynine = len(age_cat_count["35 to 39"])
fourtyplus = len(age_cat_count["40+"])

In [21]:
#percentage of players per age group
ten_p = round((ten/total)*100,2)
fourteen_p = round((fourteen/total)*100,2)
nineteen_p = round((nineteen/total)*100,2)
twentyfour_p = round((twentyfour/total)*100,2)
twentynine_p = round((twentynine/total)*100,2)
thirtyfour_p = round((thirtyfour/total)*100,2)
thirtynine_p = round((thirtynine/total)*100,2)
fourtyplus_p = round((fourtyplus/total)*100,2)

In [22]:
#Create a summary data frame to hold the results
age_demographic_summary = pd.DataFrame({
                                  "Age group":["<10","10 to 14","15 to 19", "20 to 24","25 to 29", "30 to 34", "35 to 39", "40+"],
                                  "Total count":[ten,fourteen,nineteen,twentyfour,twentynine,thirtyfour,thirtynine,fourtyplus],
                                "Percentage of players":[ten_p,fourteen_p,nineteen_p,twentyfour_p,twentynine_p,thirtyfour_p,thirtynine_p,fourtyplus_p]
    
})
age_demographic_summary

Age group  Total count  Percentage of players
0       <10           17                   2.95
1  10 to 14           22                   3.82
2  15 to 19          107                  18.58
3  20 to 24          258                  44.79
4  25 to 29           77                  13.37
5  30 to 34           52                   9.03
6  35 to 39           31                   5.38
7       40+           12                   2.08

# Purchasing Analysis (Age)





* Optional: give the displayed data cleaner formatting


* Display the summary data frame

#Bin the purchase_data data frame by age
purchase_data.sort_values(by = "Age", ascending = True)

In [23]:
#total sum of purchase ($) per age group TOTAL PURCHASE VALUE
total_purchase_by_age = group_age_category["Price"].sum()
total_purchase_by_age

Age category
<10           77.13
10 to 14      82.78
15 to 19     412.89
20 to 24    1114.06
25 to 29     293.00
30 to 34     214.00
35 to 39     147.67
40+           38.24
Name: Price, dtype: float64

### Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below

In [24]:
#total number of perchases per age group PURCHASE COUNT
total_num_purchases_by_age = purchase_data["Age category"].value_counts()
total_num_purchases_by_age

20 to 24    365
15 to 19    136
25 to 29    101
30 to 34     73
35 to 39     41
10 to 14     28
<10          23
40+          13
Name: Age category, dtype: int64

In [25]:
#AVERAGE PURCHASE PRICE per age group

ave_purch_ten = round(total_purchase_by_age["<10"]/total_num_purchases_by_age["<10"],2)
ave_purch_fourteen = round(total_purchase_by_age["10 to 14"]/total_num_purchases_by_age["10 to 14"],2)
ave_purch_nineteen = round(total_purchase_by_age["15 to 19"]/total_num_purchases_by_age["15 to 19"],2)
ave_purch_twenfour = round(total_purchase_by_age["20 to 24"]/total_num_purchases_by_age["20 to 24"],2)
ave_purch_twennine = round(total_purchase_by_age["25 to 29"]/total_num_purchases_by_age["25 to 29"],2)
ave_purch_thirfour = round(total_purchase_by_age["30 to 34"]/total_num_purchases_by_age["30 to 34"],2)
ave_purch_thirnine = round(total_purchase_by_age["35 to 39"]/total_num_purchases_by_age["35 to 39"],2)
ave_purch_fourtyplus = round(total_purchase_by_age["40+"]/total_num_purchases_by_age["40+"],2)
ave_purch_fourtyplus

2.94

In [26]:
#AVERAGE TOTAL PURCHASE AMOUNT PER PERSON

ave_purch_person_ten = round(total_purchase_by_age["<10"]/ten,2)
ave_purch_person_fourteen = round(total_purchase_by_age["10 to 14"]/fourteen,2)
ave_purch_person_nineteen = round(total_purchase_by_age["15 to 19"]/nineteen,2)
ave_purch_person_twentyfour = round(total_purchase_by_age["20 to 24"]/twentyfour,2)
ave_purch_person_twentynine = round(total_purchase_by_age["25 to 29"]/twentynine,2)
ave_purch_person_thirtyfour = round(total_purchase_by_age["30 to 34"]/thirtyfour,2)
ave_purch_person_thirtynine = round(total_purchase_by_age["35 to 39"]/thirtynine,2)
ave_purch_person_fourtyplus = round(total_purchase_by_age["40+"]/fourtyplus,2)

### Create a summary data frame to hold the results

In [27]:
purchase_analysis_age_df = pd.DataFrame({
    
                                    "Age Ranges":["<10","10 to 14","15 to 19", "20 to 24","25 to 29", "30 to 34", "35 to 39", "40+"],
                                    "Purchase Count":[23,28,136,365,101,73,41,13],
                                    "Average Purchase Price":[ave_purch_ten,ave_purch_fourteen,ave_purch_nineteen,ave_purch_twenfour,
                                                              ave_purch_twennine,ave_purch_thirfour,ave_purch_thirnine,ave_purch_fourtyplus],
                                    "Total Purchase Value":[77.13,82.78,412.89,1114.06,293.00,214.00,147.67,38.24],
                                    "Avg Total Purchase per Person":[ave_purch_person_ten,ave_purch_person_fourteen,ave_purch_person_nineteen,
                                                                    ave_purch_person_twentyfour,ave_purch_person_twentynine,ave_purch_person_thirtyfour,
                                                                    ave_purch_person_thirtynine,ave_purch_person_fourtyplus]
})

purchase_analysis_age_df

Age Ranges  Purchase Count  Average Purchase Price  Total Purchase Value  \
0        <10              23                    3.35                 77.13   
1   10 to 14              28                    2.96                 82.78   
2   15 to 19             136                    3.04                412.89   
3   20 to 24             365                    3.05               1114.06   
4   25 to 29             101                    2.90                293.00   
5   30 to 34              73                    2.93                214.00   
6   35 to 39              41                    3.60                147.67   
7        40+              13                    2.94                 38.24   

   Avg Total Purchase per Person  
0                           4.54  
1                           3.76  
2                           3.86  
3                           4.32  
4                           3.81  
5                           4.12  
6                           4.76  
7                           3.19

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [28]:
#Purchase Count
#finding total maximum purchase by one player 
top_group = purchase_data.groupby('SN').sum()
top_group.sort_values("Price", ascending = False)

Purchase ID  Age  Item ID  Price
SN                                           
Lisosia93           1630  125      442  18.96
Idastidru52         1999   96      527  15.45
Chamjask73          1306   66      339  13.83
Iral74              2285   84      518  13.62
Iskadarya95          713   60      321  13.10
...                  ...  ...      ...    ...
Ililsasya43          702   19       12   1.02
Irilis75             582   20       19   1.02
Aidai61              282   21      155   1.01
Chanirra79           586   23      155   1.01
Alo38                 63   20      125   1.00

[576 rows x 4 columns]

In [29]:
top_group.loc[:,"Price"].max()

top_group.loc[top_group["Price"] == 18.96, :]
top_purchase_name = "Lisosia93"
top_purchase_price = 18.96


In [30]:
#PURCHASE COUNT
purchases_per_player = purchase_data["SN"].value_counts()
purchases_per_player


Lisosia93      5
Iral74         4
Idastidru52    4
Lisim78        3
Iskadarya95    3
              ..
Seolollo93     1
Eolan54        1
Jeyciman68     1
Ardonmol96     1
Idacal95       1
Name: SN, Length: 576, dtype: int64

In [31]:
#PURCHASE COUNT
purchases_count_df = pd.DataFrame(purchases_per_player)
purchases_count_df

SN
Lisosia93     5
Iral74        4
Idastidru52   4
Lisim78       3
Iskadarya95   3
...          ..
Seolollo93    1
Eolan54       1
Jeyciman68    1
Ardonmol96    1
Idacal95      1

[576 rows x 1 columns]

In [32]:
#TOTAL PURCHASE VALUE
purchase_sum_per_player = top_group["Price"]
total_purchase_value = purchase_sum_per_player.sort_values(ascending = False)
total_purchase_value_df = pd.DataFrame(total_purchase_value)
total_purchase_value_df

Price
SN                
Lisosia93    18.96
Idastidru52  15.45
Chamjask73   13.83
Iral74       13.62
Iskadarya95  13.10
...            ...
Irilis75      1.02
Isurria36     1.02
Chanirra79    1.01
Aidai61       1.01
Alo38         1.00

[576 rows x 1 columns]

### Average purchase price


In [33]:
ave_purch_pp = total_purchase_value_df["Price"]/purchases_count_df["SN"]
ave_purch_price_df = pd.DataFrame(ave_purch_pp)
ave_purch_price_df

0
Adairialis76   2.280000
Adastirin33    4.480000
Aeda94         4.910000
Aela59         4.320000
Aelaria33      1.790000
...                 ...
Yathecal82     2.073333
Yathedeu43     3.010000
Yoishirrala98  4.580000
Zhisrisu83     3.945000
Zontibe81      2.676667

[576 rows x 1 columns]

In [34]:
top_summary_df = purchases_count_df.merge(ave_purch_price_df, left_index = True, right_index = True)

In [35]:
top_summary_df = top_summary_df.merge(total_purchase_value_df, left_index = True, right_index = True)
top_summary_df

SN         0  Price
Lisosia93     5  3.792000  18.96
Iral74        4  3.405000  13.62
Idastidru52   4  3.862500  15.45
Lisim78       3  3.340000  10.02
Iskadarya95   3  4.366667  13.10
...          ..       ...    ...
Seolollo93    1  3.410000   3.41
Eolan54       1  1.540000   1.54
Jeyciman68    1  2.050000   2.05
Ardonmol96    1  1.550000   1.55
Idacal95      1  2.070000   2.07

[576 rows x 3 columns]

### Display a preview of the summary data frame

In [36]:
top_summary_df = top_summary_df.rename(columns = {"SN":"Purchase Count",0:"Average Purchase Price","Price":"Total Purchase Value"})
top_summary_df[["Average Purchase Price","Total Purchase Value"]].astype(float)
top_summary_df["Average Purchase Price"] = top_summary_df["Average Purchase Price"].map("${:.2f}".format)
top_summary_df["Total Purchase Value"] = top_summary_df["Total Purchase Value"].map("${:.2f}".format)

top_summary_df.head()

Purchase Count Average Purchase Price Total Purchase Value
Lisosia93                 5                  $3.79               $18.96
Iral74                    4                  $3.40               $13.62
Idastidru52               4                  $3.86               $15.45
Lisim78                   3                  $3.34               $10.02
Iskadarya95               3                  $4.37               $13.10

# Most Popular Items






* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame

In [37]:
#Retrieve the Item ID, Item Name, and Item Price columns
item_name_id_price = purchase_data[["Item ID","Item Name","Price"]]
item_name_id_price

Item ID                                  Item Name  Price
0        108  Extraction, Quickblade Of Trembling Hands   3.53
1        143                          Frenzied Scimitar   1.56
2         92                               Final Critic   4.88
3        100                                Blindscythe   3.27
4        131                                       Fury   1.44
..       ...                                        ...    ...
775       60                                       Wolf   3.54
776      164                           Exiled Doomblade   1.63
777       67      Celeste, Incarnation of the Corrupted   3.46
778       92                               Final Critic   4.19
779       50                                       Dawn   4.60

[780 rows x 3 columns]

In [38]:
#Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value
group_id_name = item_name_id_price.groupby(["Item ID","Item Name"])

In [39]:
#total purchase value
total_purchase_value = group_id_name.sum()
final_summary_df = total_purchase_value.rename(columns = {"Price":"Total Purchase Value"})
final_summary_df

Total Purchase Value
Item ID Item Name                                                         
0       Splinter                                                      5.12
1       Crucifer                                                     11.77
2       Verdict                                                      14.88
3       Phantomlight                                                 14.94
4       Bloodlord's Fetish                                            8.50
...                                                                    ...
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76
179     Wolf, Promise of the Moonwalker                              26.88
181     Reaper's Toll                                                 8.30
182     Toothpick                                                    12.09
183     Dragon's Greatsword                                           3.27

[179 rows x 1 columns]

In [40]:
#purchase count
x = purchase_data.groupby(["Item ID"])["Item Name"].value_counts()
print(x)
purchase_count_df2 = pd.DataFrame(x).rename(columns={"Item Name":"Purchase Count"})
purchase_count_df2 

Item ID  Item Name                                   
0        Splinter                                         4
1        Crucifer                                         4
2        Verdict                                          6
3        Phantomlight                                     6
4        Bloodlord's Fetish                               5
                                                         ..
178      Oathbreaker, Last Hope of the Breaking Storm    12
179      Wolf, Promise of the Moonwalker                  6
181      Reaper's Toll                                    5
182      Toothpick                                        3
183      Dragon's Greatsword                              3
Name: Item Name, Length: 179, dtype: int64


Purchase Count
Item ID Item Name                                                   
0       Splinter                                                   4
1       Crucifer                                                   4
2       Verdict                                                    6
3       Phantomlight                                               6
4       Bloodlord's Fetish                                         5
...                                                              ...
178     Oathbreaker, Last Hope of the Breaking Storm              12
179     Wolf, Promise of the Moonwalker                            6
181     Reaper's Toll                                              5
182     Toothpick                                                  3
183     Dragon's Greatsword                                        3

[179 rows x 1 columns]

In [41]:
final_summary_df = final_summary_df.merge(purchase_count_df2,left_index = True, right_index = True)
final_summary_df

Total Purchase Value  \
Item ID Item Name                                                            
0       Splinter                                                      5.12   
1       Crucifer                                                     11.77   
2       Verdict                                                      14.88   
3       Phantomlight                                                 14.94   
4       Bloodlord's Fetish                                            8.50   
...                                                                    ...   
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76   
179     Wolf, Promise of the Moonwalker                              26.88   
181     Reaper's Toll                                                 8.30   
182     Toothpick                                                    12.09   
183     Dragon's Greatsword                                           3.27   

                                                      Purchase Count  
Item ID Item Name                                                     
0       Splinter                                                   4  
1       Crucifer                                                   4  
2       Verdict                                                    6  
3       Phantomlight                                               6  
4       Bloodlord's Fetish                                         5  
...                                                              ...  
178     Oathbreaker, Last Hope of the Breaking Storm              12  
179     Wolf, Promise of the Moonwalker                            6  
181     Reaper's Toll                                              5  
182     Toothpick                                                  3  
183     Dragon's Greatsword                                        3  

[179 rows x 2 columns]

In [42]:
#Individual Item Price
indi_item_price = round(final_summary_df["Total Purchase Value"]/final_summary_df["Purchase Count"],2)
final_summary_df["Item Price"] = indi_item_price

In [43]:
final_summary_df.sort_values("Purchase Count", ascending = False).head()

Total Purchase Value  \
Item ID Item Name                                                            
92      Final Critic                                                 59.99   
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76   
145     Fiery Glass Crusader                                         41.22   
132     Persuasion                                                   28.99   
108     Extraction, Quickblade Of Trembling Hands                    31.77   

                                                      Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                      Item Price  
Item ID Item Name                                                 
92      Final Critic                                        4.61  
178     Oathbreaker, Last Hope of the Breaking Storm        4.23  
145     Fiery Glass Crusader                                4.58  
132     Persuasion                                          3.22  
108     Extraction, Quickblade Of Trembling Hands           3.53


# Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [44]:
final_summary_df.sort_values("Total Purchase Value", ascending = False).head()

Total Purchase Value  \
Item ID Item Name                                                            
92      Final Critic                                                 59.99   
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76   
82      Nirvana                                                      44.10   
145     Fiery Glass Crusader                                         41.22   
103     Singed Scalpel                                               34.80   

                                                      Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                      Item Price  
Item ID Item Name                                                 
92      Final Critic                                        4.61  
178     Oathbreaker, Last Hope of the Breaking Storm        4.23  
82      Nirvana                                             4.90  
145     Fiery Glass Crusader                                4.58  
103     Singed Scalpel                                      4.35

# Identify three observable trends based on the data.

Out of all genders, males have the highest number of players at 484 players out of a total of 576 players that are included in this dataset. Males also had the highest number of purchases at 652 out of 780 total purchases with a male total purchase value of $1967.64 out of $2379.77 spent by all genders 

20-24 age group have the highest number of players at 1114.06 players and the highest total purchase count. This maybe due to the age group having more of an interest in games than other age groups, as well as having access to disposable income and more free time. 

The Final Critic is the most purchased item with 13 total purchases, followed by Oathbreaker , Last Hope of the Breaking Storm with 12 total purchases. 